### DISCUSSION TITLE
How to determine and change my classroom's timezone?

### DISCUSSION THREAD

**Discussion by ItaiBEdu** [community/community#65218] [GitHub Education]
When creating a GitHub Classroom assignment it is possible to set a submission deadline, as seen in the following screenshot.

![GitHubClassroomDateAndTime](https://github.com/community/community/assets/143313920/9f1370e0-9f4a-4f95-9633-d3a05fff8d95)

What is the date and time's timzone? How can I view the applicable timezone? How can I set and change the timezone?

**Comment by arafajs** [comment]
The applicable timezone is the timezone that you have set in your GitHub account. You can view the applicable timezone by going to your GitHub profile and clicking on the Settings tab. In the Settings tab, scroll down to the Time zone section and you will see the timezone that is currently set for your account.

To set or change the timezone for your GitHub account, go to your GitHub profile and click on the Settings tab. In the Settings tab, scroll down to the Time zone section and select the timezone that you want to use. Click on the Save button to save your changes.

Here are the steps on how to view and change the timezone for GitHub Classroom assignment submission deadline:

- Go to the GitHub Classroom assignment that you want to change the deadline for.
- Click on the Settings tab.
- In the Deadline section, the timezone is displayed next to the Time zone label.
- To change the timezone, click on the Time zone dropdown menu and select the timezone that you want to use.
- Click on the Save button.

**Reply by ItaiBEdu** [reply]
After some experimenting I realized that in order to make the Time zone textbox visible I must check the "Display current local time" checkbox. This makes me wonder if by setting the timezone this way the assignment deadline is calculated relative to this timezone, or if it merely makes the selected timezone display in my profile page. Even if setting my timezone this way affects the way the deadline is calculated, I don't understand why in order to set the timezone I must display it in my profile. Why can't I set the timezone secretly?

**Comment by lwclement** [comment]
I have a similar problem.  After students accept their assignment the due date is given using UTC time rather than the time zone that our university is in (Pacific - US).  This could be really confusing to some students and I would like to make sure that the due date that appears on the page students see after they accept an assignment uses a time zone that is familiar to them. 

How can I do that?

**Reply by studyvin** [reply]
I am having the same issue. I have updated my time zone in my account with no effect.

**Reply by brainark1** [reply]
I'm also seeing this issue - it's especially bad for my assignments as they're due at midnight EST, which means the day-of-the-month displayed in the due date is wrong. For example, an assignment that is due Feb 2nd, 11:59 PM EST will display as due "Feb 3rd, 4:59 AM UTC". While those are technically the same time, for students who've never worked with timezones and likely don't know what UTC even is, it will cause a lot of confusion.

**Reply by SalenaAshton** [reply]
I can set the location in my profile and in the classroom profile, but it does not save. Furthermore, I am unable to set the timezone or choose to 'display local time' for the classroom or by assignment. 

GitHub Classroom is a great introduction to GitHub and version control for my machine learning students-- but I don't like telling them 'GitHub is great except for the quirks.' I want them to enjoy using GitHub. Please resolve this issue :) 

Thanks!

**Comment by ematth** [comment]
Any updates to this issue? I've set my account and GH class's timezone, but there still doesn't appear to be a way to switch out of UTC for due dates. Any leads would be appreciated.

**Reply by vjbutt** [reply]
This bug is still present as of 26 February 2025.  It is very annoying having the deadline displayed to students in a different time-zone.

**Reply by rhanulewicz** [reply]
Still dealing with this. 1 Sep 2025.

**Reply by mrquaternion** [reply]
Same in Feb 23, 2026
